pkg/frontend/session.go

```diff
@@ -1155,10 +1155,15 @@ func (th *TxnHandler) CommitTxn() error {
 	defer cancel()
 	txnOp := th.GetTxnOperator()
 	if err := storage.Commit(ctx, txnOp); err != nil {
+		storage.Rollback(ctx, txnOp)
```

Comment:
Can these two Rollback all necessary?   Can they get combined?

Reply:
Precommit equals a new process, and I'm not sure it's possible